### PR TITLE
[GEOT-6048] Integrate Jiffle as a raster process/rendering transformation

### PR DESCRIPTION
**Pull request by aaime**
*Please do not merge* (it requires a new jai-ext release to compile)
The pull request has been made available to allow early review of the code changes.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -94,7 +94,7 @@
     <test.args></test.args>
     <src.output>${basedir}/target</src.output>
     <imageio.ext.version>1.1.24</imageio.ext.version>
-    <jaiext.version>1.0.22</jaiext.version>
+    <jaiext.version>1.0-SNAPSHOT</jaiext.version>
```
Comment (dromagnoli) [inline pom.xml]: I guess that was for testing.
make sure to update it to proper version once the new jai-ext is released with jiffle update.

File: modules/unsupported/process-raster/src/main/java/org/geotools/process/raster/JiffleProcess.java [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+            ParameterValue<?> bands =
+                    readerParams.parameter(AbstractGridFormat.BANDS.getName(null));
+            if (bands == null) {
+                LOGGER.log(Level.FINE, "The reader does not support band selection, reading all");
+                return params;
+            }
+
+            // cannot do anything if we cannot access the sample model
+            if (!(reader instanceof GridCoverage2DReader)) {
+                LOGGER.log(Level.FINE, "The reader does is not a 2D one, reading all bands");
```
[the remark this answers is not in the document]
Comment (aaime) [reply modules/unsupported/process-raster/src/main/java/org/geotools/process/raster/JiffleProcess.java]: Thanks

File: modules/unsupported/process-raster/src/main/java/org/geotools/process/raster/JiffleProcess.java [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+                return params;
+            }
+            GridCoverage2DReader r2d = (GridCoverage2DReader) reader;
+            SampleModel sampleModel = r2d.getImageLayout().getSampleModel(null);
```
[the remark this answers is not in the document]
Comment (aaime) [reply modules/unsupported/process-raster/src/main/java/org/geotools/process/raster/JiffleProcess.java]: Yeah, better safe than sorry, I'll add an extra null check

File: pom.xml (hunk 1 above)
Comment (aaime) [reply pom.xml]: Yep

**Comment by aaime** [comment]
Unrelated failure in WMTS module, the service it's trying to contact is indeed unreachable now:

```
INFO: Setting the GribIndexCache folder to: /home/travis/build/geotools/geotools/modules/plugin/coverage-multidim/grib/target/test-classes/org/geotools/coverage/io/grib/test-data/cache
2018-06-26T14:12:00.578+0000  SEVERE  Failed to execute request https://sampleserver6.arcgisonline.com/arcgis/rest/services/Toronto/ImageServer/WMTS/1.0.0/WMTSCapabilities.xml?REQUEST=GetCapabilities&VERSION=1.0.0&SERVICE=WMTS
Tests run: 5, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 33.734 sec <<< FAILURE! - in org.geotools.data.wmts.client.WMTSServiceTest
testDefaultStyleRequired(org.geotools.data.wmts.client.WMTSServiceTest)  Time elapsed: 31.011 sec  <<< ERROR!
java.net.SocketTimeoutException: Read timed out
	at java.net.SocketInputStream.socketRead0(Native Method)
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116)
	at java.net.SocketInputStream.read(SocketInputStream.java:171)
	at java.net.SocketInputStream.read(SocketInputStream.java:141)
	at sun.security.ssl.InputRecord.readFully(InputRecord.java:465)
	at sun.security.ssl.InputRecord.read(InputRecord.java:503)
	at sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:983)
	at sun.security.ssl.SSLSocketImpl.readDataRecord(SSLSocketImpl.java:940)
	at sun.security.ssl.AppInputStream.read(AppInputStream.java:105)
	at java.io.BufferedInputStream.fill(BufferedInputStream.java:246)
	at java.io.BufferedInputStream.read1(BufferedInputStream.java:286)
	at java.io.BufferedInputStream.read(BufferedInputStream.java:345)
	at sun.net.www.http.HttpClient.parseHTTPHeader(HttpClient.java:735)
	at sun.net.www.http.HttpClient.parseHTTP(HttpClient.java:678)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1587)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1492)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:263)
	at org.geotools.data.ows.SimpleHttpClient$SimpleHTTPResponse.<init>(SimpleHttpClient.java:169)
	at org.geotools.data.ows.SimpleHttpClient.get(SimpleHttpClient.java:98)
	at org.geotools.data.ows.AbstractOpenWebService.internalIssueRequest(AbstractOpenWebService.java:443)
	at org.geotools.data.ows.AbstractOpenWebService.issueRequest(AbstractOpenWebService.java:462)
	at org.geotools.data.ows.AbstractOpenWebService.negotiateVersion(AbstractOpenWebService.java:273)
	at org.geotools.data.ows.AbstractOpenWebService.<init>(AbstractOpenWebService.java:117)
```

**Pull request merged**